### PR TITLE
Viewport+transform fix

### PR DESCRIPTION
**Pull request by vyivel**
Partially fixes https://github.com/swaywm/sway/issues/6322

### CODE REVIEW

File: include/wlr/util/box.h [hunk 1]
```diff
@@ -78,4 +78,9 @@ bool wlr_box_empty(const struct wlr_box *box);
 void wlr_box_transform(struct wlr_box *dest, const struct wlr_box *box,
 	enum wl_output_transform transform, int width, int height);
 
+/**
+ * Transforms a floating-point box inside a (0, 0, width, height) box.
+ */
+void wlr_fbox_transform(struct wlr_fbox *dest, const struct wlr_fbox *box,
+	enum wl_output_transform transform, int width, int height);
```
Comment (emersion) [inline include/wlr/util/box.h:85]: We should probably turn `width` and `height` into `double` here.

**Review by emersion — COMMENTED** [review]
Apart from this nit, LGTM!

File: include/wlr/util/box.h (hunk 1 above)
Comment (vyivel) [reply include/wlr/util/box.h]: Makes sense. Done.

**Review by emersion — APPROVED** [review]
Thanks for the fix!

**Pull request merged**